lib/utils/utils.go

```diff
@@ -250,6 +250,196 @@ func SplitHostPort(hostname string) (string, string, error) {
 	return host, port, nil
 }
 
+type ProxyParams struct {
+	Host      string
+	Port      string
+	Cluster   string
+	Namespace string
+	NodeID    string
+}
+
+// HostPort constructs a value of the form `<hostname>:<port>`.  If using UUID based
+// routing, this value is likely not actually resolvable since hostname is defined as
+// `<uuid>.<cluster>` in that case.
+func (p *ProxyParams) HostPort() string {
+	port := p.Port
+	if port == "" {
+		port = "0"
+	}
+	if p.NodeID != "" && p.Cluster != "" {
+		return fmt.Sprintf("%s.%s:%s", p.NodeID, p.Cluster, port)
+	}
+	return fmt.Sprintf("%s:%s", p.Host, port)
+}
+
+func (p *ProxyParams) Encode() string {
```

Comment:
The more I think about it, I think trying to add connecting via UUID into our current proxy subsystem is a mistake.

The current system is designed around connecting via `host:port` which is 99% of the use-cases and is going to really be the only one that works well with OpenSSH (where URL encoding probably won't be supported) and not something we can drop support for.

All the special casing to support UUID seems to indicate we're going down the wrong path.

What do you think about this: we leave the current proxy subsystem as-is and we add another subsystem `proxy-uuid`. When tsh connects via UUID it just requests the `proxy-uuid` subsystem. Parsing the old subsystem won't change at all and parsing the UUID subsystem should be much simpler.

What do you think?

Reply:
I'm honestly not sure... there are pros and cons to both.

In the short term, keeping the old subsystem separate is probably the easier choice.  Definitely reduces the changes of accidentally introducing a regression.

Long-term, separate subsystems feels like more maintenance and more places for tests to accidentally miss an edge case. In terms of `openssh`, this new format seems just as compatible.  Either way, the user is manually encoding the subsystem call (see original comment).

@klizhentas any thoughts on this?